src/devices/Common/Iot/Device/Common/WeatherHelper.cs

```diff
@@ -0,0 +1,104 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Iot.Units;
+using System;
+
+namespace Iot.Device.Common
+{
+    /// <summary>
+    /// Helpers for weather. Formulas taken from http://www.reahvac.com/tools/humidity-formulas/.
+    /// </summary>
+    public class WeatherHelper
+    {
+        /// <summary>
+        /// The heat index (or apparent temperature) is used to measure the amount of discomfort
+        /// during the summer months when heat and humidity often combine to make it feel hotter
+        /// than it actually is. The heat index is usually used for afternoon high temperatures.
+        /// Use the summer simmer index instead for overnight low temperatures.
+        /// </summary>
+        /// <param name="airTemperature">The dry air temperature</param>
+        /// <param name="relativeHumidity">The relative humidity (RH) expressed as a percentage</param>
+        /// <returns>The heat index, also known as the apparent temperature</returns>
+        public static Temperature HeatIndex(Temperature airTemperature, double relativeHumidity)
+        {
+            double tf = airTemperature.Fahrenheit;
+            double rh = relativeHumidity;
+            double tf2 = Math.Pow(tf, 2);
+            double rh2 = Math.Pow(rh, 2);
+            return Temperature.FromFahrenheit((-42.379)
+                + (2.04901523 * tf)
+                + (10.14333127 * rh)
+                - (0.22475541 * tf * rh)
+                - (6.83783 * Math.Pow(10, -3) * tf2)
+                - (5.481717 * Math.Pow(10, -2) * rh2)
+                + (1.22874 * Math.Pow(10, -3) * tf2 * (rh))
+                + (8.5282 * Math.Pow(10, -4) * tf * rh2)
+                - (1.99 * Math.Pow(10, -6) * tf2 * rh2));
+        }
+
+        /// <summary>
+        /// The summer simmer index is used to measure the amount of discomfort during the summer months
+        /// when heat and humidity often combine to make it feel hotter than it actually is. The summer
+        /// simmer index is usually used for overnight low temperatures. Use the heat index instead for
+        /// afternoon high temperatures.
+        /// </summary>
+        /// <param name="airTemperature">The dry air temperature</param>
+        /// <param name="relativeHumidity">The relative humidity (RH) expressed as a percentage</param>
+        /// <returns>The summer simmer index</returns>
+        public static Temperature SummerSimmerIndex(Temperature airTemperature, double relativeHumidity)
+        {
+            double tf = airTemperature.Fahrenheit;
+            double rh = relativeHumidity;
+            return Temperature.FromFahrenheit(1.98 * (tf - ((0.55 - (0.0055 * rh)) * (tf - 58.0))) - 56.83);
```

Comment:
I'm surprised this is calculated from Fahrenheit since most of the scientific articles use Kelvin or Celsius. Please double check this is correct.

Reply:
When I created and used the unit tests, I put the values from online calculators before writing the formulas (TDD). The tests passed.